Entity/FieldType.php

```diff
@@ -724,4 +726,21 @@ protected function deserializeProperty(string $name, $value)
                 parent::deserializeProperty($name, $value);
         }
     }
+
+    public function filterDisplayOptions(DataFieldType $dataFieldType)
+    {
+        $optionsResolver = new OptionsResolver();
+        $dataFieldType->configureOptions($optionsResolver);
+        $defineOptions = $optionsResolver->getDefinedOptions();
+        $defineOptions[] = 'label';
+
+        $filtered = array_filter(
+            $this->getDisplayOptions(),
+            function ($value) use ($defineOptions) {
+                return in_array($value, $defineOptions);
+            },
+            ARRAY_FILTER_USE_KEY
```

Comment:
Using this filter option indicates you might need aan object to handle stuff. Cfr. https://matthiasnoback.nl/2020/01/rules-for-working-with-arrays-and-custom-collection-classes-in-php/ However I should look into your getDisplayOptions function first before judging!

Reply:
I notice that this is saved as a Json array. Which we can keep. But getDisplayOptions should pass that complete option array to the constructor of a DisplayOptions object which knows what options are available and makes the code cleaner wherever displayOptions is used.